test/integration/acceptance/get_account_asset_txs_test.cpp

```diff
@@ -0,0 +1,461 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "framework/integration_framework/integration_test_framework.hpp"
+#include "framework/specified_visitor.hpp"
+#include "integration/acceptance/acceptance_fixture.hpp"
+#include "utils/query_error_response_visitor.hpp"
+
+using namespace shared_model;
+using namespace integration_framework;
+using namespace shared_model::interface::permissions;
+
+class AccountAssetTxsFixture : public AcceptanceFixture {
+ public:
+  std::unique_ptr<IntegrationTestFramework> itf_;
+
+  AccountAssetTxsFixture()
+      : kSecondDomain("second"),
+        kSpectator("spectator"),
+        kCloseSpectatorId(kSpectator + "@" + kDomain),
+        kRemoteSpectatorId(kSpectator + "@" + kSecondDomain),
+        kSpectatorKeypair(
+            crypto::DefaultCryptoAlgorithmType::generateKeypair()) {}
+
+  /**
+   * Prepare state of ledger:
+   * - create accounts of target user, close and remote spectators (close
+   *   spectator - another user from the same domain as the domain of target
+   *   user account, remote - a user from domain different to domain of target
+   *   user account).
+   * - execute transfer asset from admin to target account
+   * - execute transfer asset from target to admin account
+   *
+   * @param target_permissions - set of query permissions for target account
+   * @param spectator_permissions - set of query permisisons for spectators'
+   * accounts
+   * @return reference to ITF
+   */
+  IntegrationTestFramework &prepareState(
+      const interface::RolePermissionSet &target_permissions = {},
+      const interface::RolePermissionSet &spectator_permissions = {}) {
+    auto full_target_permissions = target_permissions;
+    full_target_permissions.set(Role::kReceive);
+    full_target_permissions.set(Role::kTransfer);
+    full_target_permissions.set(Role::kAddAssetQty);
+    full_target_permissions.set(Role::kSubtractAssetQty);
+
+    // Add asset to admin and transfer to target account
+    auto prepare_tx_1 = complete(
```

Comment:
Can you please give a little more meaningful names to variables `prepare_tx_{1,2}`? Something like, as comments state, `add_transfer_asset_tx` and `transfer_asset_back_tx`

Reply:
Yes, it was a concern. That is why there are comments above for both transactions.
The idea is to have two preparation transactions. Thanks for noting.